info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java

```diff
@@ -281,21 +334,53 @@ private void storeElectionContestCandidateInDatabase(JsonObject candidate,
     Entity candidateEntity =
         Entity.newBuilder(candidateKey)
             .set("name", name)
-            .set("partyAffiliation", party + " Party")
+            .set("party", party + " Party")
+            .set("email", email)
+            .set("phone", phone)
+            .set("PhotoUrl", photoUrl)
+            .set("candidateUrl", candidateUrl)
+            .set("twitter", twitter)
             .build();
     datastore.put(candidateEntity);
     candidateIds.add(StringValue.newBuilder(Long.toString(candidateId)).build());
-    candidateIncumbency.add(BooleanValue.newBuilder(false).build());
-
+    boolean isincumbent = map.get(position).contains(name);
```

Comment:
You should also check whether the position is a key in the map first or else you'll get a 'NullPointerException' when calling `.contains()` on it.

```suggestion
    boolean isIncumbent = incumbents.containsKey(position) && incumbents.get(position).contains(name);
```